packages/node_modules/@webex/internal-plugin-devices/test/integration/spec/devices.js

```diff
@@ -412,8 +521,341 @@ describe('plugin-devices', () => {
+              () => devices.refresh()
+                .then(() =>
+                  assert.calledWith(devices.request, sinon.match.hasNested(
+                    'body.ttl', devices.config.ephemeralDeviceTTL
+                  ))));
+          });
+
+          describe('when the refresh request is sent', () => {
+            let customHeaders;
+            let customBody;
+
+            beforeEach('configure devices plugin', () => {
+              customHeaders = {
+                testHeader: 'example header value'
+              };
+
+              customBody = {
+                testBody: 'example body value'
+              };
+            });
+
+            it('should allow for custom header key:values', () => {
+              assert.isTrue(devices.registered);
+
+              devices.config.headers = {...customHeaders};
+
+              return devices.refresh()
+                .then(() =>
+                  assert.calledWith(devices.request, sinon.match.hasNested(
+                    'headers.testHeader', customHeaders.testHeader
+                  )));
+            });
+
+            it('should allow for custom body key:values', () => {
+              assert.isTrue(devices.registered);
+
+              devices.config.body = {...customBody};
+
+              return devices.refresh()
+                .then(() =>
+                  assert.calledWith(devices.request, sinon.match.hasNested(
+                    'body.testBody', customBody.testBody
+                  )));
+            });
+
+            it('should use the device\'s url property', () => {
+              assert.isTrue(devices.registered);
+
+              devices.config.body = {...customBody};
+
+              return devices.refresh()
+                .then(() =>
+                  assert.calledWith(devices.request, sinon.match.has(
+                    'uri', devices.url
+                  )));
+            });
+
+            it('should send a PUT request', () => {
+              assert.isTrue(devices.registered);
+
+              devices.config.body = {...customBody};
+
+              return devices.refresh()
+                .then(() =>
+                  assert.calledWith(devices.request, sinon.match.has(
+                    'method', 'PUT'
+                  )));
+            });
+          });
+
+          describe('when the device is successfully refreshes', () => {
+            beforeEach('setup stubs', () => {
+              devices.processRegistrationSuccess = sinon.stub();
+            });
+
+            it('should return a resolved promise',
+              () => assert.isFulfilled(devices.refresh()));
+
+            it('should call \'processRegistrationSuccess()\'', () => devices.refresh()
+              .then(() => assert.called(devices.processRegistrationSuccess)));
+          });
+
+          describe('when the device fails to refresh', () => {
+            beforeEach('setup \'register()\' stub', () => {
```

Comment:
Is this stub only used in the `should attempt to register a new device` test from line 668? If so, it probably can be moved to within the `it` block.

Reply:
Moved it to the specific `it` statement it's exclusive to.